src/export/ExportFFmpeg.cpp

```diff
@@ -715,58 +807,64 @@ static int encode_audio(AVCodecContext *avctx, AVPacket *pkt, int16_t *audio_sam
+
+      for (ch = 0; ch < avctx->GetChannels(); ch++) {
+         for (i = 0; i < nb_samples; i++) {
+            switch(static_cast<AudacityAVSampleFormat>(avctx->GetSampleFmt())) {
+            case AUDACITY_AV_SAMPLE_FMT_U8:
+               ((uint8_t*)(frame->GetData(0)))[ch + i*channelsCount] = audio_samples[ch + i*channelsCount]/258 + 128;
                break;
-            case AV_SAMPLE_FMT_U8P:
-               ((uint8_t*)(frame->data[ch]))[i] = audio_samples[ch + i*avctx->channels]/258 + 128;
+            case AUDACITY_AV_SAMPLE_FMT_U8P:
+               ((uint8_t*)(frame->GetData(ch)))[i] = audio_samples[ch + i*channelsCount]/258 + 128;
                break;
-            case AV_SAMPLE_FMT_S16:
-               ((int16_t*)(frame->data[0]))[ch + i*avctx->channels] = audio_samples[ch + i*avctx->channels];
+            case AUDACITY_AV_SAMPLE_FMT_S16:
+               ((int16_t*)(frame->GetData(0)))[ch + i*channelsCount] = audio_samples[ch + i*channelsCount];
                break;
-            case AV_SAMPLE_FMT_S16P:
-               ((int16_t*)(frame->data[ch]))[i] = audio_samples[ch + i*avctx->channels];
+            case AUDACITY_AV_SAMPLE_FMT_S16P:
+               ((int16_t*)(frame->GetData(ch)))[i] = audio_samples[ch + i*channelsCount];
                break;
-            case AV_SAMPLE_FMT_S32:
-               ((int32_t*)(frame->data[0]))[ch + i*avctx->channels] = audio_samples[ch + i*avctx->channels]<<16;
+            case AUDACITY_AV_SAMPLE_FMT_S32:
+               ((int32_t*)(frame->GetData(0)))[ch + i*channelsCount] = audio_samples[ch + i*channelsCount]<<16;
                break;
-            case AV_SAMPLE_FMT_S32P:
-               ((int32_t*)(frame->data[ch]))[i] = audio_samples[ch + i*avctx->channels]<<16;
+            case AUDACITY_AV_SAMPLE_FMT_S32P:
+               ((int32_t*)(frame->GetData(ch)))[i] = audio_samples[ch + i*channelsCount]<<16;
                break;
-            case AV_SAMPLE_FMT_FLT:
-               ((float*)(frame->data[0]))[ch + i*avctx->channels] = audio_samples[ch + i*avctx->channels] / 32767.0;
+            case AUDACITY_AV_SAMPLE_FMT_FLT:
+               ((float*)(frame->GetData(0)))[ch + i*channelsCount] = audio_samples[ch + i*channelsCount] / 32767.0;
                break;
-            case AV_SAMPLE_FMT_FLTP:
-               ((float*)(frame->data[ch]))[i] = audio_samples[ch + i*avctx->channels] / 32767.;
+            case AUDACITY_AV_SAMPLE_FMT_FLTP:
+               ((float*)(frame->GetData(ch)))[i] = audio_samples[ch + i*channelsCount] / 32767.;
                break;
-            case AV_SAMPLE_FMT_NONE:
-            case AV_SAMPLE_FMT_DBL:
-            case AV_SAMPLE_FMT_DBLP:
-            case AV_SAMPLE_FMT_NB:
+            default:
                wxASSERT(false);
                break;
             }
          }
       }
    }
 
-   pkt->data = NULL; // packet data will be allocated by the encoder
-   pkt->size = 0;
+   pkt->ResetData();
+
+   ret = ffmpeg.avcodec_encode_audio2(
```

Comment:
I notice this function, too, is "deprecated"
https://ffmpeg.org/doxygen/3.4/group__lavc__encoding.html#gae4b5ba4d50b1264a5cf5408ff32fb622

Reply:
Yes, but send/receive family is not supported in 2.2.2. When encode_audio/decode_audio will be removed - we will have to add 2 implementations.